### ISSUE TITLE
Alternate web API to get user posts

### ISSUE THREAD

**Issue by antiops** [bashonly/yt-dlp-TTUser#1]
There's a new endpoint I started seeing today through the web app. It's accessible in a tab labeled as 'Creator videos' next to the comments tab when viewing a video accessed through the user page.

It currently doesn't require any signatures or valid tokens (like the `verifyFp`) to return data

### Minimum Params
Latest: [https://www.tiktok.com/api/creator/item_list/?aid=1988&type=1&count=15&cursor=1689909282235&secUid=MS4wLjABAAAAys5jIEEWJXF-bCt_11wTP363Xvnma4SAex4cTxY-EDrUe2CPXUr1rh-tZtqa5Src&verifyFp=verify_](https://www.tiktok.com/api/creator/item_list/?aid=1988&type=1&count=15&cursor=1689909282235&secUid=MS4wLjABAAAAys5jIEEWJXF-bCt_11wTP363Xvnma4SAex4cTxY-EDrUe2CPXUr1rh-tZtqa5Src&verifyFp=verify_)

Next Page: [https://www.tiktok.com/api/creator/item_list/?aid=1988&type=1&count=15&cursor=1677186594000&secUid=MS4wLjABAAAAys5jIEEWJXF-bCt_11wTP363Xvnma4SAex4cTxY-EDrUe2CPXUr1rh-tZtqa5Src&verifyFp=verify_](https://www.tiktok.com/api/creator/item_list/?aid=1988&type=1&count=15&cursor=1677186594000&secUid=MS4wLjABAAAAys5jIEEWJXF-bCt_11wTP363Xvnma4SAex4cTxY-EDrUe2CPXUr1rh-tZtqa5Src&verifyFp=verify_)

- `secUid`
- `aid`
  - `1988`
- `type`
  - `1` returns sorted item lists on all requests
  - `0` returns item lists with dupes unless its the most recent 15 posts
- `count`
  - Max `15`
- `cursor`
  - Setting to current time in milis will return the most recent posts
  - The 'next' cursor used on frontend is the timestamp of the last post in the item list
  - Setting to `0` will return the oldest posts
- `verifyFp`
  - Any string is valid as long as it starts with `verify_`

### Res

- No next/previous cursor values
- `hasMoreLatest` & `hasMorePrevious` = true/false

### Normal params

```
https://www.tiktok.com/api/creator/item_list/?aid=1988&app_language=en&app_name=tiktok_web&browser_language=en-US&browser_name=Mozilla&browser_online=true&browser_platform=Win32&browser_version=<>&channel=tiktok_web&cookie_enabled=true&count=15&cursor=<>&device_id=<>&device_platform=web_pc&focus_state=true&from_page=video&history_len=4&is_fullscreen=false&is_page_visible=true&language=en&os=windows&priority_region=US&referer=&region=US&screen_height=1080&screen_width=1920&secUid=<>&type=0&tz_name=<>&verifyFp=<>&webcast_language=en
```

**Comment by bashonly** [comment]
Nice find. I've written a version of the user extractor that uses the new endpoint and ditches the playwright dependency; take a look / try it out:
https://github.com/bashonly/yt-dlp-TTUser/tree/creator

It has a problem, though:
The first API request (with the cursor as current timestamp in milliseconds) works fine; it returns the 15 most recent videos posted by the user.
But when the next page is requested with the cursor set to the timestamp of the final video from the current page, that request yields only 5 new videos and 10 duplicates from the previous page. This happens for every subsequent page request. 
Not sure what do to about this. I don't like the idea of assuming that the first 10 videos of each subsequent page are dupes and skipping them, since that could result in non-dupes being skipped. 
Hoping that the web API is doing something that I'm missing and this is an easy fix.

**Comment by antiops** [comment]
> Hoping that the web API is doing something that I'm missing and this is an easy fix.

Looks like its just the `type` param that I missed on subsequent calls. Setting it to `1` on all requests will return properly ordered items. Setting it to `1` on the first request also gives the same result as `0`

All that changes by setting to `1` is the `hasMoreLatest` value it removed, but its not needed since we're only going down the post list

**Comment by bashonly** [comment]
Right on, pushed working implementation to master

**Comment by antiops** [comment]
Awesome, I can confirm it's working perfectly in multiple regions.

Now we wait and see if/when the tiktok ninjas find this and neuter it.

**Comment by bashonly** [comment]
Closed by 9919edfd34f2de32ac83126a1883574fa977fb67
